src/main/java/org/jenkinsci/plugins/badge/StatusImage.java

```diff
@@ -41,7 +41,7 @@
     private final String length;
 
     StatusImage(String fileName) throws IOException {
-        etag = Jenkins.RESOURCE_PATH+'/'+fileName;
+        etag = '"' + Jenkins.RESOURCE_PATH + '/' + fileName + '"';
```

Comment:
`fileName` is one from [a small restricted set of names](https://github.com/jenkinsci/embeddable-build-status-plugin/blob/8f19e79bad2d5ee4fbca24ec1b927d52bc37647f/src/main/java/org/jenkinsci/plugins/badge/ImageResolver.java#L39-L53).  `Jenkins.RESOURCE_PATH` is a URL fragment that looks like `/static/301f1991`.  Neither contains `"` characters, so I think this patch is sufficient.

The file contents change only when somebody commits a tweak/bugfix (like d637d1cdd46a853ce2739835bebff816e9076bcc).  [Jenkins.RESOURCE_PATH](http://javadoc.jenkins-ci.org/jenkins/model/Jenkins.html#RESOURCE_PATH) changes [when Jenkins version changes, or when Jenkins is restarted](https://github.com/jenkinsci/jenkins/blob/b8f5342c73843b530ccbcd8bb3748d8365b9d762/core/src/main/java/jenkins/model/Jenkins.java#L4120-L4131) -- but stays the same if the plugin version changes.  Thus, it's possible to get stale cached images if someone upgrades just the plugin, but not Jenkins itself, and hashing just the filename won't fix that.  OTOH how likely is that someone will upgrade the plugin but not restart Jenkins?  I think we can ignore that case.